testing/scenario_app/ios/Scenarios/ScenariosUITests/iPadGestureTests.m

```diff
@@ -62,24 +62,77 @@ - (void)testPointerButtons {
+  XCTAssertTrue([app.textFields[@"5,PointerChange.down,d=2,b=2"] waitForExistenceWithTimeout:1],
                 @"PointerChange.down event did not occur for a right-click");
-  XCTAssertTrue([app.textFields[@"PointerChange.up:2"] waitForExistenceWithTimeout:1],
+  XCTAssertTrue([app.textFields[@"6,PointerChange.up,d=2,b=2"] waitForExistenceWithTimeout:1],
                 @"PointerChange.up event did not occur for a right-click");
 }
+
+- (void)testPointerHover {
+  BOOL supportsPointerInteraction = NO;
+  SEL supportsPointerInteractionSelector = @selector(supportsPointerInteraction);
+  if ([XCUIDevice.sharedDevice respondsToSelector:supportsPointerInteractionSelector]) {
+    supportsPointerInteraction =
+        performBoolSelector(XCUIDevice.sharedDevice, supportsPointerInteractionSelector);
+  }
+  XCTSkipUnless(supportsPointerInteraction, "Device does not support pointer interaction.");
+  XCUIApplication* app = [[XCUIApplication alloc] init];
+  app.launchArguments = @[ @"--pointer-events" ];
+  [app launch];
+
+  NSPredicate* predicateToFindFlutterView =
+      [NSPredicate predicateWithBlock:^BOOL(id _Nullable evaluatedObject,
+                                            NSDictionary<NSString*, id>* _Nullable bindings) {
+        XCUIElement* element = evaluatedObject;
+        return [element.identifier hasPrefix:@"flutter_view"];
+      }];
+  XCUIElement* flutterView = [[app descendantsMatchingType:XCUIElementTypeAny]
+      elementMatchingPredicate:predicateToFindFlutterView];
+  if (![flutterView waitForExistenceWithTimeout:kSecondsToWaitForFlutterView]) {
+    NSLog(@"%@", app.debugDescription);
+    XCTFail(@"Failed due to not able to find any flutterView with %@ seconds",
+            @(kSecondsToWaitForFlutterView));
+  }
+
+  XCTAssertNotNil(flutterView);
+
+  SEL hover = @selector(hover);
+  XCTAssertTrue([flutterView respondsToSelector:hover],
+                @"If supportsPointerInteraction is true, this should be true too.");
+  [flutterView performSelector:hover];
+  [NSThread sleepForTimeInterval:1.0];
+  XCTAssertTrue([app.textFields[@"0,PointerChange.add,d=0,b=0"] waitForExistenceWithTimeout:1],
+                @"PointerChange.add event did not occur for a hover");
+  [flutterView tap];
+  XCTAssertTrue([app.textFields[@"1,PointerChange.add,d=1,b=0"] waitForExistenceWithTimeout:1],
+                @"PointerChange.add event did not occur for a tap");
+  XCTAssertTrue([app.textFields[@"2,PointerChange.down,d=1,b=0"] waitForExistenceWithTimeout:1],
+                @"PointerChange.down event did not occur for a tap");
+  XCTAssertTrue([app.textFields[@"3,PointerChange.up,d=1,b=0"] waitForExistenceWithTimeout:1],
+                @"PointerChange.up event did not occur for a tap");
+  XCTAssertTrue([app.textFields[@"4,PointerChange.remove,d=0,b=0"] waitForExistenceWithTimeout:1],
+                @"The hover pointer was not removed after a tap");
```

Comment:
Was this sent before the change you made, or is it actually testing the change?  Can we get in a scenario where you are sending 2 removes now?

Reply:
This is the change, there was no remove sent before, the hover pointer was not removed at any point